concordium-consensus/src/Concordium/GRPC2.hs

```diff
@@ -439,6 +562,9 @@ decodeAccountIdentifierInput 1 dta = do
 decodeAccountIdentifierInput 2 dta = AccIndex . AccountIndex <$> peek (castPtr dta)
 decodeAccountIdentifierInput n _ = error $ "Unknown account identifier tag: " ++ show n
 
+decodeModuleRefInput :: Ptr Word8 -> IO ModuleRef
```

Comment:
I don't think you are checking on the Rust end that the moduleRef is actually 32 bytes. So this function is problematic.

For block hashes and account addresses we check on the Rust end that they are of correct length, so we can then do this.

Reply:
I have added a check on the rust side now